syncs/syncs.go

```diff
@@ -152,3 +153,67 @@ func (s Semaphore) TryAcquire() bool {
 func (s Semaphore) Release() {
 	<-s.c
 }
+
+// Map is a Go map protected by a sync.RWMutex.
+// It is preferred over [sync.Map] for maps with entries that change
+// at a relatively high frequency.
+type Map[K comparable, V any] struct {
+	mu sync.RWMutex
+	m  map[K]V
+}
+
+func (m *Map[K, V]) Load(key K) (value V, ok bool) {
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+	value, ok = m.m[key]
+	return value, ok
+}
+
+func (m *Map[K, V]) Store(key K, value V) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	if m.m == nil {
```

Comment:
Not that it matters much here but you could save a few lines by using mak.Set but that has a trade off of someone having to look into the package for implementation.

Reply:
Done.